src/globals/operations/findOne.ts

```diff
@@ -45,38 +41,22 @@ async function findOne<T extends Record<string, unknown>>(args: Args): Promise<T
     accessResult = await executeAccess({ req }, globalConfig.access.read);
   }
 
-  const query = await Model.buildQuery({
-    where: combineQueries({ globalType: { equals: slug } }, accessResult),
-    payload,
-    locale,
-    overrideAccess,
-    globalSlug: slug,
-  });
-
   // /////////////////////////////////////
   // Perform database operation
   // /////////////////////////////////////
 
-  let doc = await Model.findOne(query).lean() as any;
-
-  if (!doc) {
-    doc = {};
-  } else if (doc._id) {
-    doc.id = doc._id;
-    delete doc._id;
-  }
-
-  doc = JSON.stringify(doc);
-  doc = JSON.parse(doc);
-  doc = sanitizeInternalFields(doc);
+  let doc = await req.payload.db.findGlobal({
+    slug,
+    locale,
+    where: overrideAccess ? { globalType: { equals: slug } } : combineQueries({ globalType: { equals: slug } }, accessResult),
```

Comment:
@DanRibbens please review if the way I'm handling `overrideAccess` makes sense here

Reply:
this is accurate